subgraphs/aave-governance/protocols/aave-governance/src/aave-governance-v-2.ts

```diff
@@ -0,0 +1,142 @@
+import { GovernanceStrategy } from "../../../generated/AaveGovernanceV2/GovernanceStrategy";
+import { GovernanceFramework } from "../../../generated/schema";
+
+export function handleProposalCanceled(event: ProposalCanceled): void {
+  _handleProposalCanceled(event.params.id.toString(), event);
+}
+
+export function handleProposalCreated(event: ProposalCreated): void {
+  let executor = event.params.executor;
+  let quorumVotes = getQuorumFromContract(
+    event.address,
+    executor,
+    event.block.number.minus(BIGINT_ONE)
+  );
+
+  _handleProposalCreated(
+    event.params.id.toString(),
+    event.params.creator.toHexString(),
+    executor.toHexString(),
+    event.params.targets,
+    event.params.values,
+    event.params.signatures,
+    event.params.calldatas,
+    event.params.startBlock,
+    event.params.endBlock,
+    event.params.ipfsHash,
+    quorumVotes,
+    event
+  );
+}
+
+export function handleProposalExecuted(event: ProposalExecuted): void {
+  _handleProposalExecuted(event.params.id.toString(), event);
+}
+
+export function handleProposalQueued(event: ProposalQueued): void {
+  _handleProposalQueued(event.params.id.toString(), event.params.executionTime);
+}
+
+export function handleVoteEmitted(event: VoteEmitted): void {
+  let proposal = getOrCreateProposal(event.params.id.toString());
+
+  // if state is pending (i.e. the first vote), set state, quorum, delegates and tokenholders
+  if (proposal.state == ProposalState.PENDING) {
+    proposal.state = ProposalState.ACTIVE;
+    // Set snapshot for quorum, tokenholders and delegates
+    proposal.quorumVotes = getQuorumFromContract(
+      event.address,
+      Address.fromString(proposal.executor),
+      event.block.number.minus(BIGINT_ONE)
+    );
+    let governance = getGovernance();
+    proposal.tokenHoldersAtStart = governance.currentTokenHolders;
+    proposal.delegatesAtStart = governance.currentDelegates;
+  }
+
+  // Proposal will be updated as part of handler
+  _handleVoteEmitted(
+    proposal,
+    event.params.voter.toHexString(),
+    event.params.votingPower,
+    event.params.support,
+    event
+  );
+}
+
+// VotingDelayChanged (newVotingDelay, initiatorChange)
+export function handleVotingDelayChanged(event: VotingDelayChanged): void {
+  let governanceFramework = getGovernanceFramework(event.address.toHexString());
+  governanceFramework.votingDelay = event.params.newVotingDelay;
+  governanceFramework.save();
+}
+
+// Helper function that imports and binds the contract
+function getGovernanceFramework(contractAddress: string): GovernanceFramework {
+  let governanceFramework = GovernanceFramework.load(contractAddress);
+
+  if (!governanceFramework) {
+    governanceFramework = new GovernanceFramework(contractAddress);
+    let contract = AaveGovernanceV2.bind(Address.fromString(contractAddress));
+    governanceFramework.name = contract.NAME();
+    governanceFramework.type = GOVERNANCE_NAME;
```

Comment:
The `type` in the framework is different than our previous OZ subgraphs

We should have an OZ-specific const e.g. `OZ_FRAMEWORK_TYPE` but then use a new `AAVE_V2_FRAMEWORK_TYPE` (or something similar, not strongly opinionated on name) for here

Reply:
ooo I see, Aave governance has its own constant file

The governance name should ideally be unique to each governance entity (Aave in this case) and the framework type const shared for all framework entities using said framework.

This is something I see we may be doing wrong in the OZ implementations as well

governance_name != governance_framework_type

Something worth discussing offline

Reply:
Isn't that exactly what we're doing here?
 governanceFramework.name is the name from the contract i.e. `Aave Governance v2` or `dYdX Governance`
 governanceFramework.type is our constant name for the type `AaveGovernanceV2`
 
 I think only confusion is the variable name `GOVERNANCE_NAME` should be `GOVERNANCE_TYPE`